### PR TITLE
Simplify metadata processing

### PR DESCRIPTION
**Pull request by turian**
This mainly gets rid of the name "process metadata", which was weird and made no sense.

### CODE REVIEW

File: heareval/tasks/dcase2016_task2.py [hunk 1]
```diff
@@ -117,10 +117,7 @@ def get_requires_metadata(self, split: str) -> pd.DataFrame:
             )
             metadata = metadata.assign(
                 relpath=sound_file,
-                slug=lambda df: df.relpath.apply(self.slugify_file_name),
-                subsample_key=self.get_subsample_key,
                 split=lambda df: split,
-                split_key=self.get_split_key,
```
Comment (turian) [inline heareval/tasks/dcase2016_task2.py:123]: I accidentally remove split_key in this PR. I fix it in https://github.com/neuralaudio/hear-preprocess/pull/8

File: heareval/tasks/pipeline.py [hunk 2]
```diff
@@ -308,8 +309,15 @@ def split_train_test_val(self, metadata: pd.DataFrame):
...
-        process_metadata = self.get_process_metadata()
+        # Process metadata gets all metadata to be used for the task
+        metadata = self.get_all_metadata()
+
+        metadata.reset_index(drop=True)
+        metadata = metadata.assign(
+            slug=lambda df: df.relpath.apply(self.slugify_file_name),
+            subsample_key=self.get_subsample_key,
+            split_key=self.get_split_key,
+        )
```
Comment (jorshi) [inline heareval/tasks/pipeline.py:320]: Nice

File: heareval/tasks/pipeline.py [hunk 3]
```diff
@@ -237,23 +220,41 @@ def get_subsample_key(df: DataFrame) -> Series:
         assert "slug" in df, "slug column not found in the dataframe"
         return df["slug"].apply(str).apply(filename_to_int_hash)
 
-    def get_process_metadata(self) -> pd.DataFrame:
+    @staticmethod
+    def get_stratify_key(df: DataFrame) -> Series:
         """
-        Return a dataframe containing the task metadata for this
-        entire task.
+        CURRENTLY UNUSED.
```
Comment (jorshi) [inline heareval/tasks/pipeline.py:226]: Comment on why? -- This stratification works well for multi-class scene based tasks, but becomes more complicated for multilabel and event based tasks, so this is unused for the time being until we can figure that out. Or something like that.

**Review by jorshi — APPROVED** [review]
This is great -- super clean

File: heareval/tasks/pipeline.py (hunk 3 above)
Comment (turian) [reply heareval/tasks/pipeline.py:224]: @jorshi I just straight up remove it later so I'll merge it. Maybe worth a talk in our summary paper

**Pull request merged**